Edu benefits confirmation #2994

alexose opened the pull request:
Adds a confirmation page.

The data is placeholder data for now-- Will update when the claims API returns the necessary data.

alexose committed:
navigation and beginnings

alexose committed:
Merge branch 'master' into edu-benefits-confirmation-#2994

alexose committed:
submitmessage markup and styles

alexose committed:
improve styles

alexose committed:
improve address display

alexose committed:
Merge branch 'master' into edu-benefits-confirmation-#2994

alexose committed:
Merge branch 'edu-benefits-confirmation-#2994' of github.com:department-of-veterans-affairs/vets-website into edu-benefits-confirmation-#2994

jbalboni commented on src/js/edu-benefits/containers/SubmitPage.jsx:
```diff
@@ -0,0 +1,42 @@
+      <SubmitMessage
+          claimType="Education Benefits"
+          confirmation="8-910fks-76"
+          date="Jul 12, 2016"
```
I though we were getting the date back from the api now?

jbalboni commented on src/js/edu-benefits/containers/SubmitPage.jsx:
```diff
@@ -0,0 +1,42 @@
+          date="Jul 12, 2016"
+          address={address}
+          claimedBenefits="Chapter 33"
+          relinquishedBenefits="Chapter 1607"/>
```
Could we pull these from the form data we have? They're both in the form on the benefits selection page.

alexose replied:
We could, although I'd rather see all of this information come back from the server.  If that proves to be tricky, we can definitely pull it from the store.

jbalboni commented on src/js/edu-benefits/containers/SubmitPage.jsx:
```diff
@@ -0,0 +1,42 @@
+
+function mapStateToProps(state, ownProps) {
+  return {
+    currentLocation: ownProps.location.pathname
```
Do we need this?

alexose replied:
Nooope!

jbalboni commented on src/js/rx/containers/Main.jsx:
```diff
@@ -12,15 +12,13 @@ class Main extends React.Component {
-      );
-    }
+    alertBox = (
+      <AlertBox
```
Is it ok for us to use this? There are some css classes prefixed with rx in the code for it.

alexose replied:
It seems like a useful component to have.  I'd rather fix it in `./common` than reinvent the wheel... I think I'll do that

jbalboni commented on src/js/edu-benefits/containers/SubmitPage.jsx:
```diff
@@ -0,0 +1,42 @@
+class SubmitPage extends React.Component {
+  render() {
+		// TODO: shared address component?
+    const address = (
```
I think we'd want to get this from the api eventually, too? I think region depends on some data in the form.

jbalboni replied:
Looks like we'll be getting a `processedAt` property, so we'd probably use that somehow.

jbalboni reviewed (approved):
This looks good to me. I think there's some more data we could fill in from the form and had some other minor questions.

alexose committed:
Merge branch 'master' into edu-benefits-confirmation-#2994

alexose committed:
remove unneeded prop

alexose committed:
make alertbox less rx-specific.  still figuring out what's wrong with svgs

alexose committed:
fix svgs

alexose commented:
@U-DON I changed out the rx-alert prefixes on the AlertBox component so that we could use it in edu-benefits.  Hopefully that's all right with you!

I also fixed the problem with SVGs not loading ( #2930 ).  The problem was... weird.  In short, webpack was trying to inline them incorrectly.  I added a module that does it the right way, and it seems to be working now:

![image](https://cloud.githubusercontent.com/assets/627264/18602929/1db9b1be-7c22-11e6-9f50-912b3e027eb4.png)

alexose committed:
update shrinkwrap

alexose committed:
fix svg url loader config

alexose committed:
Merge branch 'master' into edu-benefits-confirmation-#2994

U-DON commented on src/js/rx/containers/Main.jsx:
```diff
@@ -12,15 +12,13 @@ class Main extends React.Component {
-            status={this.props.alert.status}/>
-      );
-    }
+    alertBox = (
```
Why doesn't this have the conditional anymore? Is its visibility triggered by something else now?

alexose replied:
Oops, didn't mean to commit that.  Reverted.

alexose committed:
Merge branch 'master' into edu-benefits-confirmation-#2994

alexose committed:
Merge branch 'master' into edu-benefits-confirmation-#2994

alexose committed:
Merge branch 'edu-benefits-confirmation-#2994' of github.com:department-of-veterans-affairs/vets-website into edu-benefits-confirmation-#2994

alexose committed:
revert change to rx

bshyong committed:
remove unneeded dependency

alexose committed:
Merge branch 'master' into remove-shallow-compare

alexose committed:
remove shallow-compare from package.json and rebuild shrinkwrap

alexose committed:
attempt to update shrinkwrap again

alexose committed:
Merge branch 'master' into edu-benefits-confirmation-#2994
